src/test/java/uk/co/ramp/OutbreakTest.java

```diff
@@ -97,54 +97,63 @@ public void runContactDataWithRandoms() {
             .filter(status -> status == SUSCEPTIBLE)
             .count();
 
-    // we expect this to roughly follow an exp decay
-    double test = popSize * Math.exp(-randomInfection * days);
-    Assert.assertEquals(test / (double) popSize, sus / (double) popSize, 0.1);
+    // We expect this to roughly follow an exp decay, but with a large tolerance.
+    //
+    // If r='randomInfection' ratio of people become other status than SUSCEPTIBLE,
+    // after T days, remaining population of SUSCEPTIBLE persons should be
+    // (1-r)^T = exp(T log(1-r))
+    // Previously, an approximation log(1-r)=r was applied but this is valid only
+    // when r is small
+    // double test = popSize * Math.exp(Math.log(1. - randomInfection) * days);
+
+    // Calibrate the error not in the population itself but in the logarithmic rate space.
+    Assert.assertEquals(
+        -Math.log(sus / (double) popSize) / days, -Math.log(1 - randomInfection), 0.1);
   }
 
```

Comment:
@vinopm This is one possible alternative. In the current test, approximately 0.12 and 0.1 are the values inside the first and second argements in `Asset.assertEquals`, respectively. We can use even smaller tolerance if using geometric rate. Another alternative is to use tolerance proportional to randomInfection, say, `0.5 * randomInfection`. However this is problematic because even if `randomInfection=0` virus diffuses purely by contact and number of `SUSCEPTIBLE` persons decline over time. So using a constant rather than proportional one is commendable.

Reply:
Thanks for the explanation RIkiya. So we could change the tolerance to say 0.05?

Reply:
Yes, if you have happy I will set it into 0.05. Also will remove commented-out old codes for simplification.

Reply:
Sure, I am happy with that